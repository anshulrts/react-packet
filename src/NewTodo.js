import React, { Component } from 'react';
import './NewTodo.css';

class NewTodo extends Component {
    constructor(props) {
        super(props);
        this.state = { item: '' };
        this.handleUpdate = this.handleUpdate.bind(this);
        // This is the local addTodo
        this.addTodo = this.addTodo.bind(this);
    }
    render() { 
        return ( 
            <div className="NewTodo">
                <input type="text" onChange={ this.handleUpdate } value={ this.state.item } /> &nbsp; &nbsp;
                <button onClick={this.addTodo}>Add</button>
            </div>
        );
    }

    handleUpdate(event) {
        this.setState({ item: event.target.value });
    }

    addTodo() {
        this.props.addTodo(this.state.item);
        this.setState({item : ''});
    }
}
 
export default NewTodo;