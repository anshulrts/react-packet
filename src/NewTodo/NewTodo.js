import React, { Component } from 'react';
import './NewTodo.css';
import { Button, Input, InputGroup } from "reactstrap";

class NewTodo extends Component {
    constructor(props) {
        super(props);
        this.state = { item: '' };
        this.handleUpdate = this.handleUpdate.bind(this);
        // This is the local addTodo
        this.addTodo = this.addTodo.bind(this);
    }
    render() { 
        return ( 
            <div className="NewTodo">
                <InputGroup>
                    <Input
                        type="text"
                        onChange = { this.handleUpdate }
                        value = { this.state.item }
                        placeholder = "Input item name goes here..."
                    />
                    <Button
                        onClick = { this.addTodo }
                        color = "primary"
                    >
                        Add
                    </Button>
                </InputGroup>
            </div>
        );
    }

    handleUpdate(event) {
        this.setState({ item: event.target.value });
    }

    addTodo() {
        this.props.addTodo(this.state.item);
        this.setState({item : ''});
    }
}
 
export default NewTodo;