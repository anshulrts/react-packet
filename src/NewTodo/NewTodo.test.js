import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import NewTodo from './NewTodo';

describe(NewTodo, () => {
    const mockAddTodo = jest.fn();
    const component = shallow(<NewTodo addTodo={mockAddTodo} />);

    it("renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<NewTodo addTodo={mockAddTodo} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    
    it("renders a NewTodo component", () => {
        expect(component.contains(<div className="NewTodo" />));
    });

    it("renders and matched our snapshot", () => {
        const component = renderer.create(<NewTodo addtodo={mockAddTodo} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("contains the form", () => {
        expect(component.find("Input")).toHaveLength(1);
        expect(component.find("Button")).toHaveLength(1);
    });

    it("calls the mocked addTodo function passed in when the add new button is clicked", () => {
        component.find("Button").simulate("click");
        expect(mockAddTodo).toBeCalled();
    });

    it("updates the form when keys are pressed", () => {
        const updateKey = "New Todo";
        component.instance().handleUpdate({target : { value : updateKey }});
        expect(component.state("item")).toEqual(updateKey);
    });

    it("Blanks out the Todo name when the button is clicked", () => {
        const updateKey = "I should be empty";
        component.instance().handleUpdate({target : { value: updateKey }});
        expect(component.state("item")).toEqual(updateKey);
        component.find("Button").simulate("click");
        expect(component.state("item")).toHaveLength(0);
    })

});