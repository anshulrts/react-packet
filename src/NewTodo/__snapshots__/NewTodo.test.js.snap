// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodo renders and matched our snapshot 1`] = `
<div
  className="NewTodo"
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
      
  <button
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;
